### ISSUE TITLE
Type for money

### ISSUE THREAD

**Issue by worldmind** [OAI/OpenAPI-Specification#889]
For money params we need special type, in Python, for example, it named [decimal](https://docs.python.org/3/library/decimal.html), PostgreSQL has [numeric](https://www.postgresql.org/docs/9.6/static/datatype-numeric.html#DATATYPE-NUMERIC-DECIMAL) type.

**Comment by darrelmiller** [comment]
The semantics of `decimal` can be added using the `format` property.  It will be included in the `format` registry that we will be creating https://github.com/OAI/OpenAPI-Specification/issues/845

**Comment by spacether** [comment]
For context here openapi-generator does this with:
```
type: string
format: number
```

**Comment by karenetheridge** [comment]
I could almost see a "number" format being useful, to specify strings that contain numeric data, but there's so many ways of representing a number, and some of them might not be suitable for a particular application. Also some numeric representations aren't practical in some languages.

For example, one could start out defining a number format as "any string that can be converted to a number in my language" -- so for perl, the easy way of implementing that is with [`looks_like_number`](https://perldoc.perl.org/perlapi#looks_like_number), which handles values like "2.0", "2.5", "2e-3", "Inf" and "NaN" - but doesn't handle "0xDEADBEEF" or "0x1.23p-4" or "0 but true" (which are all treated internally as numbers without warning".  Or in C, one could use `atoi()` but that doesn't match many numeric forms cleanly either.

tldr: formats like this are best defined locally to an application using a definition, so it can be tailored to the exact needs of that application, framework and language.
